### DISCUSSION TITLE
How can I reliably use rewrites at build time? Can I get the current host from Next.js (on Vercel) during build?

### DISCUSSION THREAD

**Discussion by stephenhmarsh** [vercel/next.js#55759] [Help]
### Summary

I need to fetch data to build static pages from an API that's behind a reverse proxy configured through `next.config.js`.

For example:

```js

fallback: [
  // These rewrites are checked after both pages/public files
  // and dynamic routes are checked
  {
    source: '/:path*',
    destination: `https://www.example.com/:path*`,
  },
],

```

Since the `fetch` API can't use a relative path on the backend, I need to know the full hostname and port of where the build is running so that it can go through the reverse proxy.

My builds will work locally because I can:
```js
fetch("localhost:3000/foo")
```

And `/foo` will get rewritten to `example.com/foo` and I'll successfully get the data.

But on Vercel, the best I have is `process.env.NEXT_PUBLIC_VERCEL_URL` -- but during the BUILD stage, that is apparently not online and responding to requests??  

It seems what I need to be able to do in `getStaticProps` is something like:

```js
fetch(`${an_address_for_whatever_host_is_running_during_build}/foo`)
```

**Comment by devjiwonchoi** [comment]
Just asking, so the hostname and the port may vary?

**Reply by stephenhmarsh** [reply]
I'm not sure.  The first problem is that the hostname and port are totally unknown/undocumented afaik. 

Most googling says to use `process.env.VERCEL_URL`, which I tried, and does not apparently respond to requests until after build (needs to be officially 'deployed').
